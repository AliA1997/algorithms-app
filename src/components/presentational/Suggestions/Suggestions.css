.suggestions {
    overflow: auto;
    text-align: center;
    list-style: none;
    background: lightgray;
    position: absolute;
    width: 80%;
    max-height: 20%;
    margin-top: -0.75%;
    padding: 0.25%;
}

.suggestion {
    padding: 1%;
    font-size: 1.1em;
    font-weight: 500;
    width: 100%;
}
.suggestion:hover {
    background: #fff;
}

.suggestion:active {
    background: #fff;
}