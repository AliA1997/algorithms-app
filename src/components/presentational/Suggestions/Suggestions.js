import React from 'react';
//import css file for styling
import './Suggestions.css';

export const Suggestions = (props) => {
    let options = props.suggestions.map((sugg, i) => sugg.name.toLowerCase().includes(props.suggestion) &&
    <li className='suggestion' key={i} onClick={() => props.clickSuggestion(sugg.name)}>{sugg.name}</li>).filter(s => s);
    console.log(options);
    return (
        <div className='suggestions' style={{display: props.showSuggestion ? 'inline-block': 'none'}}>{options}</div>
    );
};

export default Suggestions;