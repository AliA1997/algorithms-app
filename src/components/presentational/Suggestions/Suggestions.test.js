//import react since your importing render from react-test-renderer
import React from 'react';
//import the render from react-test-renderer
import render from 'react-test-renderer';
///import the component.
import { Suggestions } from './Suggestions';
//Define a sample prop.
const sampleProp = [
    {
        name: 'A'
    },
    {
        name: 'B'
    },
    {
        name: 'C'
    },
    {
        name: 'D'
    }
]

describe('Test my suggestions------------', () => {
    test('Test each suggestion', () => {
        const tree = render.create(<Suggestions suggestions={sampleProp} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})