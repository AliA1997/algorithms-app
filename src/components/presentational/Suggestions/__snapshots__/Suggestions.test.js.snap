// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test my suggestions------------ Test each suggestion 1`] = `
<div
  className="suggestions"
  style={
    Object {
      "display": "none",
    }
  }
/>
`;
